pqarrow/builder/optbuilders_test.go

```diff
@@ -1,18 +1,18 @@
-package builder_test
+package builder
```

Comment:
Why not keep this package as `builder_test`? I think it's good practice for testing APIs

Reply:
Can you elaborate as to how it helps test apis? To me it feels like an antipattern to have multiple packages in a directory

Reply:
It's a testing idiom (https://github.com/golang/go/issues/25223) used for black box testing of a package so that you explicitly only allow usage of the public-facing interface.